pkg/hintrunner/zero/zerohint_math.go

```diff
@@ -225,3 +225,149 @@ func createIsNNOutOfRangeHinter(resolver hintReferenceResolver) (hinter.Hinter,
 	}
 	return newIsNNOutOfRangeHint(a), nil
 }
+
+func newIsPositiveHint(value, dst hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "IsPositive",
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.math_utils import is_positive
+			//> ids.is_positive = 1 if is_positive(
+			//>     value=ids.value, prime=PRIME, rc_bound=range_check_builtin.bound) else 0
+
+			isPositiveAddr, err := dst.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			value, err := value.Resolve(vm)
+			if err != nil {
+				return err
+			}
+			valueFelt, err := value.FieldElement()
+			if err != nil {
+				return err
+			}
+
+			var v memory.MemoryValue
+			if utils.FeltIsPositive(valueFelt) {
+				v = memory.MemoryValueFromFieldElement(&utils.FeltOne)
+			} else {
+				v = memory.MemoryValueFromFieldElement(&utils.FeltZero)
+			}
+			return vm.Memory.WriteToAddress(&isPositiveAddr, &v)
+		},
+	}
+}
+
+func createIsPositiveHinter(resolver hintReferenceResolver) (hinter.Hinter, error) {
+	// This hint is executed for the negative values.
+	// If the value was non-negative, it's usually handled by the IsNN hint.
+
+	a, err := resolver.GetResOperander("a")
+	if err != nil {
+		return nil, err
+	}
+	return newIsNNOutOfRangeHint(a), nil
+}
+
+func newSplitIntAssertRangeHint(value hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "SplitIntAssertRange",
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//> assert ids.value == 0, 'split_int(): value is out of range.'
+
+			value, err := value.Resolve(vm)
+			if err != nil {
+				return err
+			}
+			valueFelt, err := value.FieldElement()
+			if err != nil {
+				return err
+			}
+			if valueFelt.IsZero() {
```

Comment:
Shouldn't we check the opposite here? If `valueFelt` is zero then the assertion is correct, so no need to return this error.

Reply:
Good catch!